src/miner_consensus_mgr.erl

```diff
@@ -912,10 +912,28 @@ do_dkg(Addrs, Artifact, Sign, Done, N, Curve, Create,
             ok = libp2p_group_relcast:handle_input(DKGGroup, start),
             lager:info("height ~p delay ~p Address: ~p, ConsensusWorker pos: ~p, Group name ~p",
                        [Height, Delay, MyAddress, Pos, DKGGroupName]),
-            {true, State#state{current_dkgs = DKGs#{{Height, Delay} => DKGGroup}}};
+            {true, State#state{current_dkgs = clean_dkgs(DKGs#{{Height, Delay} => DKGGroup})}};
         false ->
             lager:info("not in DKG this round at height ~p ~p", [Height, Delay]),
-            {false, State#state{current_dkgs = DKGs#{{Height, Delay} => out}}}
+            {false, State#state{current_dkgs = clean_dkgs(DKGs#{{Height, Delay} => out})}}
+    end.
+
+clean_dkgs(DKGs) ->
```

Comment:
What happens if this DKG "wins" after getting cleaned?

Reply:
I mean there's a reason the branch name is `limit-dkg-bad`